src/lib/components/Swap/Summary/Details.tsx

```diff
@@ -48,43 +50,48 @@ export default function Details({ trade, allowedSlippage }: DetailsProps) {
   const integrator = window.location.hostname
   const feeOptions = useAtomValue(feeOptionsAtom)
 
+  const { i18n } = useLingui()
   const details = useMemo(() => {
+    const rows = []
     // @TODO(ianlapham): Check that provider fee is even a valid list item
-    return [
-      // [t`Liquidity provider fee`, `${swap.lpFee} ${inputSymbol}`],
-      [
+
+    if (feeOptions) {
+      const parsedConvenienceFee = formatCurrencyAmount(outputAmount.multiply(feeOptions.fee), 6, i18n.locale)
+      rows.push([
         t`${integrator} fee`,
-        feeOptions &&
-          `${outputAmount.multiply(feeOptions.fee).toSignificant(2)} ${
-            outputCurrency.symbol || currencyId(outputCurrency)
-          }`,
-      ],
-      [
-        t`Price impact`,
-        `${priceImpact.toFixed(2)}%`,
-        !priceImpact.lessThan(ALLOWED_PRICE_IMPACT_HIGH)
-          ? 'error'
-          : !priceImpact.lessThan(ALLOWED_PRICE_IMPACT_MEDIUM)
-          ? 'warning'
-          : undefined,
-      ],
-      trade.tradeType === TradeType.EXACT_INPUT
-        ? [t`Maximum sent`, `${trade.maximumAmountIn(allowedSlippage).toSignificant(6)} ${inputCurrency.symbol}`]
-        : [],
-      trade.tradeType === TradeType.EXACT_OUTPUT
-        ? [t`Minimum received`, `${trade.minimumAmountOut(allowedSlippage).toSignificant(6)} ${outputCurrency.symbol}`]
-        : [],
-      [
-        t`Slippage tolerance`,
-        `${allowedSlippage.toFixed(2)}%`,
-        !allowedSlippage.lessThan(MIN_HIGH_SLIPPAGE) && 'warning',
-      ],
-    ].filter(isDetail)
+        `${parsedConvenienceFee} ${outputCurrency.symbol || currencyId(outputCurrency)}`,
+      ])
+    }
+
+    const priceImpactMessage = !priceImpact.lessThan(ALLOWED_PRICE_IMPACT_HIGH)
+      ? 'error'
+      : !priceImpact.lessThan(ALLOWED_PRICE_IMPACT_MEDIUM)
+      ? 'warning'
+      : undefined
+    rows.push([t`Price impact`, `${priceImpact.toFixed(2)}%`, priceImpactMessage])
+
+    if (trade.tradeType === TradeType.EXACT_INPUT) {
+      const localizedMaxSent = formatCurrencyAmount(trade.maximumAmountIn(allowedSlippage), 6, i18n.locale)
+      rows.push([t`Maximum sent`, `${localizedMaxSent} ${inputCurrency.symbol}`])
+    }
+
+    if (trade.tradeType === TradeType.EXACT_OUTPUT) {
+      const localizedMaxSent = formatCurrencyAmount(trade.minimumAmountOut(allowedSlippage), 6, i18n.locale)
+      rows.push([t`Minimum received`, `${localizedMaxSent} ${outputCurrency.symbol}`])
+    }
+
+    rows.push([
+      t`Slippage tolerance`,
+      `${allowedSlippage.toFixed(2)}%`,
+      !allowedSlippage.lessThan(MIN_HIGH_SLIPPAGE) && 'warning',
+    ])
+
+    return rows.filter(isDetail)
```

Comment:
If you change this to be `rows.push`, then you shouldn't need to `filter` - you just shouldn't be adding them in the first place. Either  continue to use a filtered array, or only use a dynamically created array. (Either is fine because we are memoizing - I would favor readability, and that is subjective.)